woltka/workflow.py

```diff
@@ -0,0 +1,590 @@
+        demux = demux is not False
+        if demux:
+            files = [path_]
+
+        # validate with given sample Ids
+        else:
+            sample = path2stem(path_, ext)
+            if samples and samples != [sample]:
+                raise ValueError(errmsg)
+            files = {path_: sample}
+            samples = [sample]
+
+    # path is a directory of alignment files
+    elif isdir(path_):
+
+        # turn off demultiplexing if not decided
+        demux = demux or False
+
+        # get a map of plausible sample Ids to files
+        map_ = id2file_map(path_, ext, not demux and samples)
+        if len(map_) == 0:
+            raise ValueError('No valid file found in directory.')
+        if demux:
+            files = sorted([join(path_, x) for x in map_.values()])
+
+        # validate with given sample Ids
+        else:
+            if not samples:
+                samples = sorted(map_.keys())
+            elif len(map_) < len(samples):
+                raise ValueError(errmsg)
+            files = {join(path_, map_[x]): x for x in samples}
+
+    else:
+        raise ValueError(f'"{path_}" is not a valid file or directory.')
+
+    click.echo(f'Alignment files to read: {len(files)}.')
+    click.echo(f'Demultiplexing: {"on" if demux else "off"}.')
+    return samples, files, demux
+
+
+def build_align_proc(coords_fp: str = None,
+                     overlap:   int = None) -> object:
+    """Build alignment processor.
+
+    Parameters
+    ----------
+    coords_fp : str, optional
+        Path to gene coordinates file.
+    overlap : int, optional
+        Read/gene overlapping percentage threshold
+
+    Returns
+    -------
+    object
+        Alignment processor.
+
+    Notes
+    -----
+    Currently two processors are supported: Plain() for regular alignments
+    (i.e., simple query-to-subject maps), Ordinal() for alignments with
+    coordinates which will be used to match queries (reads) and genes. The
+    presence of a gene coordinates file (`coords_fp`) is an indicator for
+    using the latter.
+    """
+    if coords_fp:
+        click.echo(f'Reading gene coordinates...', nl=False)
+        with readzip(coords_fp) as fh:
+            coords = read_gene_coords(fh, sort=True)
+        click.echo(' Done.')
+        click.echo(f'Total host sequences: {len(coords)}.')
+        return Ordinal(coords, whether_prefix(coords),
+                       overlap and overlap / 100)
+    else:
+        return Plain()
+
+
+def build_hierarchy(names_fp:   str = None,
+                    nodes_fp:   str = None,
+                    newick_fp:  str = None,
+                    lineage_fp: str = None,
+                    ranktb_fp:  str = None,
+                    map_fps:   list = None,
+                    map_rank:  bool = False) -> (dict, dict, dict, str):
+    """Construct hierarchical classification system.
+
+    Parameters
+    ----------
+    names_fp : str, optional
+        Taxonomic names file.
```

Comment:
```suggestion
        NCBI format taxonomic names file.
```

Reply:
Same as above. This `names_fp` is not necessarily NCBI-style.